pyproject.toml

```diff
@@ -32,7 +32,7 @@ python = ">=3.8,<3.12"
 nautobot = "^2.0.0"
 azure-identity = { version = "^1.15.0", optional = true }
 azure-keyvault-secrets = { version = "^4.8.0", optional = true }
-boto3 = { version = "~1.19.5", optional = true }
+boto3 = { version = "~1.34", optional = true }
```

Comment:
Wonder if we should use a `^` dependency instead of a `~` dependency - how has `boto3` behaved historically with regard to breaking changes? Is it well behaved enough that we can loosen this constraint?

Also, have you tested this?

Reply:
In general I agree on the `^` versus `~` but I opted for keeping it as it was. The reason is that I have no knowledge on how `boto3` behaves and whether it tends to introduce breaking changes in minor versions (although I'd guess no).

I have not tested this outside of the CI pipeline unittests.

Reply:
@glennmatthews any opinion on this?

Reply:
The `boto3` changelog hasn't had a change *documented* as breaking since v0.0.19, FWIW. Our usage is pretty minimal so I think it's unlikely that we'll have any issue with future 1.x versions of boto3, so personally I'd be fine with making this a `^` dependency until/unless proven otherwise.

Reply:
Changed.